aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -209,6 +209,12 @@
    */
   public static final String CLIENT_ASSUME_ROLE_REGION = "client.assume-role.region";
 
+  /**
+   * Enables eTag checks for S3 PUT and MULTIPART upload requests.
+   */
+  public static final String CLIENT_ENABLE_ETAG_CHECK = "client.enable.etag-check";
```

Comment:
Would it make more sense to use `client.etag.enable-check` or something with `etag` as the second part?

Some of the other ones seem to follow a pattern of "client".<resource>.<configuration>, which I think makes more sense.

Reply:
since this is only for s3, I think it makes more sense to use `s3` prefix. Also the convention of naming for this type of config is `xxx.enabled` (see example like `SPARK_WRITE_PARTITIONED_FANOUT_ENABLED`)

Reply:
The name "etag check" feels a bit not informative to me, people without context has no idea what this is doing. Maybe something like `s3.integrity-check.enabled` or `s3.checksum.enabled` sounds better?

Reply:
I'm a big fan of `s3` as the prefix, and then `s3.checksum.enabled` would be my preferred choice.

This really explains what the intent is, vs etag-check more explains how it works and isn't as informative for users who don't know the underlying details of S3.

Reply:
SGTM, updated to `s3.checksum.enabled`